src/app/clusters/dishwasher-alarm-server/dishwasher-alarm-server.cpp

```diff
@@ -0,0 +1,299 @@
+    BitMask<AlarmMap> mask;
+
+    status = GetMaskValue(endpoint, &mask);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        return status;
+    }
+    SendNotifyEvent(endpoint, becameActive, becameInactive, newState, mask);
+
+    return status;
+}
+
+void DishwasherAlarmServer::SendNotifyEvent(EndpointId endpointId, BitMask<AlarmMap> becameActive, BitMask<AlarmMap> becameInactive,
+                                            BitMask<AlarmMap> newState, BitMask<AlarmMap> mask)
+{
+    Events::Notify::Type event{ .active = becameActive, .inactive = becameInactive, .state = newState, .mask = mask };
+    EventNumber eventNumber;
+    CHIP_ERROR error = LogEvent(event, endpointId, eventNumber);
+    if (CHIP_NO_ERROR != error)
+    {
+        ChipLogError(Zcl, "[Notify] Unable to send notify event: %s [endpointId=%d]", error.AsString(), endpointId);
+    }
+}
+
+static Status resetHandler(const app::ConcreteCommandPath & commandPath, const chip::BitMask<AlarmMap> alarms)
+{
+    EndpointId endpoint = commandPath.mEndpointId;
+
+    EmberAfStatus status;
+    chip::BitMask<AlarmMap> state = 0;
+    status                        = State::Get(endpoint, &state);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        return Status::Failure;
+    }
+
+    chip::BitMask<AlarmMap> latch = 0;
+    status                        = Latch::Get(endpoint, &latch);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        return Status::Failure;
+    }
+
+    uint32_t alarmValue = alarms.Raw();
+    uint32_t stateValue = state.Raw();
+    uint32_t latchValue = latch.Raw();
+
+    // Flip the bits of alarms (i.e. ~alarms)
+    alarmValue = 0xFFFF ^ alarmValue;
+
+    // Reset state from active to inactive
+    stateValue = stateValue & alarmValue;
+
+    state.SetRaw(stateValue);
+    status = State::Set(endpoint, state);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        return Status::Failure;
+    }
+
+    // Reset latch from active to inactive
+    latchValue = latchValue & alarmValue;
+
+    latch.SetRaw(latchValue);
+    status = Latch::Set(endpoint, latch);
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        return Status::Failure;
+    }
+
+    return Status::Success;
+}
+
+static Status modifyEnabledHandler(const app::ConcreteCommandPath & commandPath, const chip::BitMask<AlarmMap> mask)
+{
+    EndpointId endpoint = commandPath.mEndpointId;
+    chip::BitMask<AlarmMap> support;
+
+    EmberAfStatus status = Attributes::Supported::Get(endpoint, &support);
+
+    if (status != EMBER_ZCL_STATUS_SUCCESS)
+    {
+        ChipLogProgress(Zcl, "Dishwasher Alarm: ERR: reading  supported, err:0x%x", status);
+        return Status::Failure;
+    }
+    else
+    {
+        ChipLogDetail(Zcl, "Dishwasher Alarm: Supported ep%d value: %" PRIu32 "", endpoint, support.Raw());
+    }
+
+    if ((support & mask) != mask)
+    {
+        return Status::Failure;
```

Comment:
This does not match the spec.  Test needed.

Reply:
Sure , add test already.